pkg/issuer/acme/dns/util/wait.go

```diff
@@ -77,7 +78,12 @@ func updateDomainWithCName(r *dns.Msg, fqdn string) string {
 }
 
 // checkDNSPropagation checks if the expected TXT record has been propagated to all authoritative nameservers.
-func checkDNSPropagation(fqdn, value string, nameservers []string) (bool, error) {
+func checkDNSPropagation(fqdn, value string, nameservers []string,
+	useAuthoritative bool) (bool, error) {
+	if !useAuthoritative {
+		return checkAuthoritativeNss(fqdn, value, nameservers)
```

Comment:
It's a good point and I wrestled with the correct way to do this but landed on the current solution based on:

- All strings and string slices that represent nameservers in this code assume it there is `:port` at the end. The only exception was this code so I feel making the change keeps it inline with everything else.
- By keeping checkAuthoritativeNss appending default DNS port to the nameserver slice it has two negative impacts on the code.
  1. We would need to unpack the resolver nameservers when this flag is in use to remove the port.
  2. If the user supplied their own resolvers with custom ports they would be lost and we would be trying to hit those resolvers on 53 only. Technically we could put some smarts in checkAuthoritativeNss to test if a port exists but I feel the function would now be trying to correct input data that should already be in the form the function wants.

My personal opinion is that string has probably become too much of a flexible type for nameservers now. It may be worth considering aliasing string to a nameserver type that has a function that can return the port set or a default value.